ros2doctor/ros2doctor/api/package.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from ament_index_python import get_packages_with_prefixes
+from catkin_pkg.package import InvalidPackage
+from catkin_pkg.package import parse_package
+from ros2doctor.api import DoctorCheck
+from ros2doctor.api import DoctorReport
+from ros2doctor.api import Report
+from ros2doctor.api import Result
+
+import rosdistro
+
+
+def get_ros_packages_info() -> dict:
+    """
+    Return all current distro's packages info using rosdistro API.
+
+    :return: a dictionary contains package name, release, source and status
+    """
+    distro_name = os.environ.get('ROS_DISTRO')
```

Comment:
It looks like this will raise `RuntimeError` if `ROS_DISTRO` is unset. What should `PackageCheck` and `PackageReport` do in that case?

```python3
>>> import rosdistro
>>> url = rosdistro.get_index_url()
>>> i = rosdistro.get_index(url)
>>> data = rosdistro.get_distribution(i, None).get_data()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3/dist-packages/rosdistro/__init__.py", line 114, in get_distribution
    dist_file = get_distribution_file(index, dist_name)
  File "/usr/lib/python3/dist-packages/rosdistro/__init__.py", line 119, in get_distribution_file
    data = _get_dist_file_data(index, dist_name, 'distribution')
  File "/usr/lib/python3/dist-packages/rosdistro/__init__.py", line 186, in _get_dist_file_data
    raise RuntimeError("Unknown release: '{0}'. Valid release names are: {1}".format(dist_name, ', '.join(sorted(index.distributions.keys()))))
RuntimeError: Unknown release: 'None'. Valid release names are: ardent, bouncy, crystal, dashing, eloquent, groovy, hydro, indigo, jade, kinetic, lunar, melodic, noetic
```

Reply:
I'll add `RuntimeError` to the `except` block in PackageCheck and PackageReport. In check, it will pass error message to result; in report, it will output an empty report with names only.